source/grid/tria.cc

```diff
@@ -10740,6 +10732,13 @@ namespace internal
             // decision. the following function sets these flags:
             triangulation.fix_coarsen_flags();
 
+            // first clear flags on lines, since we need them to determine
+            // which lines will be refined
+            triangulation.clear_user_flags_line();
+
+            // also clear flags on hexes, since we need them to mark those
+            // cells which are to be coarsened
+            triangulation.clear_user_flags_hex();
```

Comment:
Note that I had to switch these lines here, because we previously depended on the side effect that the `fix_coarsen_flag` function in the iteration before would clear the flags for us. Of course, that is dangerous, so it is better to do it explicitly and in the right place.